nntrainer/optimizers/optimizer_factory.cpp

```diff
@@ -22,6 +23,25 @@ static bool istrequal(const std::string &a, const std::string &b) {
   return std::equal(a.begin(), a.end(), b.begin(),
                     [](char a, char b) { return tolower(a) == tolower(b); });
 }
+
+/// helper function to convert enum to string
+/// @todo this should be integrated into appcontext
+static const std::string integer_to_string_type(const OptType &type) {
```

Comment:
I wonder why not use unordered_map?

Reply:
This is code bit is temporary before moving into appcontext. #745 will use unordered map.

Reply:
Then maybe this PR can be merged with 745? as it seems like most of this patch will be replaced with 745.

Reply:
Since #735 has to wait a lot and #745 need a lot more work...  I think it's better to merge this for #735 :)

Reply:
cool.